Scripts/Services/TextToSpeech/V1/TextToSpeechServiceExtension.cs

```diff
@@ -0,0 +1,273 @@
+        /// indicated voice. You must make the request with credentials for the instance of the service that owns the
+        /// custom model. Omit the parameter to use the specified voice with no customization. (optional).
+        /// </summary>
+        public string CustomizationId { get { return _customization_id; } set { _customization_id = value; } }
+        #endregion
+
+        #region Sessionless - Streaming
+        /// <summary>
+        /// This callback object is used by the Synthesize().
+        /// </summary>
+        /// <param name="results">The ResultList object containing the results.</param>
+        public delegate void OnSynthesize(byte[] result);
+
+
+        /// <summary>
+        /// This starts the service listening and it will invoke the callback for any recognized text.
+        /// OnSynthesize() must be called by the user to queue audio data to send to the service.
+        /// StopListening() should be called when you want to stop listening.
+        /// </summary>
+        /// <param name="callback">All recognize results are passed to this callback.</param>
+        /// <returns>Returns true on success, false on failure.</returns>
+        public bool StartListening(OnSynthesize callback)
+        {
+            if (callback == null)
+                throw new ArgumentNullException("callback");
+            if (_isListening)
+                return false;
+            if (!CreateListenConnector())
+                return false;
+
+            Dictionary<string, string> customHeaders = new Dictionary<string, string>();
+            foreach (KeyValuePair<string, string> kvp in customRequestHeaders)
+            {
+                customHeaders.Add(kvp.Key, kvp.Value);
+            }
+
+            if (customHeaders != null && _listenSocket != null)
+            {
+                foreach (KeyValuePair<string, string> kvp in customHeaders)
+                    _listenSocket.Headers.Add(kvp.Key, kvp.Value);
+            }
+
+            _isListening = true;
+            _listenCallback = callback;
+
+            return true;
+        }
+
+        /// <summary>
+        /// This function should be invoked with the text input after StartListening() method has been invoked.
+        /// The user should continue to invoke this function until they are ready to call StopListening().
+        /// </summary>
+        /// <param name="text">.</param>
+        /// <returns>True if text was sent or enqueued, false if text was discarded.</returns>
+        public bool OnListen(string text)
+        {
+            bool textSentOrEnqueued = false;
+
+            if (_isListening)
+            {
+                Log.Debug("ExampleTextToSpeechV1", "on listen: {0}", text);
+                Dictionary<string, string> data = new Dictionary<string, string>();
+                data["text"] = text;
+                data["accept"] = Accept;
+                _listenSocket.Send(new WSConnector.TextMessage(Json.Serialize(data)));
```

Comment:
Should `timings` be sent as well?
https://cloud.ibm.com/apidocs/text-to-speech/text-to-speech#websocket_methods